designs/Commands/clear-package.md

```diff
@@ -0,0 +1,92 @@
+# Clearing an specific package from NuGet cache
+
+* Status: **Incubating**
+* Author(s): [Fernando Aguilar](https://github.com/dominofire)
+
+The purpose is to add a feature to `nuget locals` command to delete a package from NuGet cache folder
+
+
+## Problem Background
+
+There's some interest on having features to delete an specific package instead of clearing the whole cache.
+
+- https://github.com/NuGet/Home/issues/5713 : Comments on deleting specific packages
+- https://github.com/NuGet/Home/issues/4980 : Comments on cache handling polices
+- https://github.com/chgill-MSFT/NuGetCleaner : A clean-up tool relying on last-acesss file metadata in NTFS file system
+
+In PackageReference world, the global packages folders are shared across all restored projects. 
+
+There's no current API or command to delete an specific package.
+
+
+## Who are the customers
+
+All `dotnet nuget locals` users.
+
+
+## Goals
+
+- Add command options to `dotnet nuget locals` to delete _one_ package from a NuGet cache folder, either _one_ or multiple versions.
+
+
+## Non-Goals
+
+- Cache expiration policies are not covered
+- Deleting cached packages from an specific folder are not covered 
+- `packages.config` projects are not covered. To delete a package, delete the pacakge folder from `packages` folder
+- `nuget.exe` commands are not covered
+
+
+## Solution Overview
+
+From ideas described in [this comment](https://github.com/NuGet/Home/issues/5713#issuecomment-320560636), the following command options will delete an specific package. Exmples are listed below.
+
+Delete a package from cache with an specific version:
+
+```
+dotnet nuget locals global-packages --clear --package a --version 1.3
+```
+
+Delete all package versions from cache:
+
+```
+dotnet nuget locals global-packages --clear --package a
+```
+
+Delete all versions of multiple packages:
+
+```
+dotnet nuget locals http-cache --clear --package a
+dotnet nuget locals http-cache --clear --package b
+dotnet nuget locals http-cache --clear --package c
+dotnet nuget locals http-cache --clear --package d
+```
+
+Delete all versions of multiple packages (version 2):
+
+```
+dotnet nuget locals global-packages -clear -packages a,b,c,d,e
```

Comment:
Other examples have used double dash `--clear` and `--package`, rather than single dash `-clear` and `-packages`.

The dotnet CLI uses the GNU argument syntax where only single character arguments use a single dash and any multi-character argument must use double dash. I don't know if the dotnet CLI implements it, but programs using GNU's `getopt` treat single dash `-ab` the same as two arguments `-a -b`, which is why `git clean -xdf` is the same as `git clean -x -d -f`, but different to `git clean --xdf`